.travis.yml

```diff
@@ -43,85 +37,18 @@ matrix:
     - env: INSTALL_TYPE=develop TEST=none FITS="astropy" TRAVIS_PYTHON_VERSION="3.5"
 
 before_install:
-  # General configurations
-  - export LIBGFORTRANVER="3.0"
-  - export SHERPA_CHANNEL=sherpa
-  - export XSPEC_CHANNEL=cxc/channel/dev
-  - export MINICONDA=/home/travis/miniconda
-
-  # Install, update, and configure conda
-  - wget http://repo.continuum.io/miniconda/Miniconda-latest-Linux-x86_64.sh -O miniconda.sh
-  - chmod +x miniconda.sh
-  - ./miniconda.sh -b -p $MINICONDA
-  - export PATH=$MINICONDA/bin:$PATH
-  - conda update --yes conda
-  - conda config --add channels ${SHERPA_CHANNEL}
-  - conda config --add channels ${XSPEC_CHANNEL}
-
-  # Figure out requested dependencies
-  - if [ -n "${MATPLOTLIBVER}" ]; then MATPLOTLIB="matplotlib=${MATPLOTLIBVER}"; fi
-  - if [ -n "${NUMPYVER}" ]; then NUMPY="numpy=${NUMPYVER}"; fi
-  - if [ -n "${XSPECVER}" ];
-     then export XSPEC="xspec-modelsonly=${XSPECVER}";
-    fi
-  - echo ${MATPLOTLIB} ${NUMPY} ${FITS} ${XSPEC}
-
-  # Create and activate conda build environment
-  # We create a new environment so we don't care about the python version in the root environment.
-  - conda create --yes --quiet -n build python=$TRAVIS_PYTHON_VERSION pip ${MATPLOTLIB} ${NUMPY} ${MKL} libgfortran=$LIBGFORTRANVER $XSPEC $FITS
-  - source activate build
+  - . travis/setup_conda.sh
 
   # XSPEC and DS9
   - if [ -n "${XSPECVER}" ];
-     then DS9_SITE=http://ds9.si.edu/download/centos6/;
-     XPA_SITE=http://ds9.si.edu/download/centos6/;
-     DS9_TAR=ds9.centos6.7.5.tar.gz;
-     XPA_TAR=xpa.centos6.2.1.17.tar.gz;
-     wget --quiet $DS9_SITE$DS9_TAR;
-     wget --quiet $XPA_SITE$XPA_TAR;
-     THIS_DIR=`pwd`;
-     cd $MINICONDA/bin; tar xf $THIS_DIR/$DS9_TAR; tar xf $THIS_DIR/$XPA_TAR; cd -;
-     sudo apt-get install -qq libwcs4 wcslib-dev libx11-dev libsm-dev libxrender-dev;
-     export HEADAS=$MINICONDA/envs/build/Xspec/spectral;
-     export LD_LIBRARY_PATH=$MINICONDA/envs/build/Xspec/x86_64-unknown-linux-gnu-libc2.15-0/lib/;
-     export XSPEC_INCLUDE_PATH=$MINICONDA/envs/build/Xspec/x86_64-unknown-linux-gnu-libc2.15-0/include/;
-     sed -i.orig s/#with-xspec=True/with-xspec=True/g setup.cfg;
-     sed -i.orig "s|#xspec_lib_dirs = None|xspec_lib_dirs=$LD_LIBRARY_PATH|g" setup.cfg;
-     sed -i.orig "s|#xspec_include_dirs = None|xspec_include_dirs=$XSPEC_INCLUDE_PATH|g" setup.cfg;
-    fi
-
-  # No test data, then remove submodule (git automatically clones recursively)
-  - if [ ${TEST} == none ];
-     then git submodule deinit -f .;
-    fi
-
-  # Install test data as a package, then remove the submodule
-  - if [ ${TEST} == package ];
-     then pip install ./sherpa-test-data;
-     git submodule deinit -f .;
+      then . travis/setup_xspec_ds9.sh;
     fi
 
 install:
-    - python setup.py $INSTALL_TYPE &> install.log
+    - python setup.py $INSTALL_TYPE
```

Comment:
Do we not have to worry about the screen output anymore (i.e. the removal of the pipe-to install.log) part? I forget exactly why this was added in the first place.

Reply:
Oops, good catch, I was supposed to put the redirect back in after debugging the build. It's interesting that no jobs failed, though. A lot of times the jobs would fail for the logs were too long. Not sure if they removed the limitation, but we don't really need to see all the output from the build, so I'll add the redirect back.